src/test/mir-opt/const-promotion-extern-static/rustc.BAR.PromoteTemps.diff

```diff
@@ -0,0 +1,64 @@
+  // MIR for `BAR`
+  // source = MirSource { instance: Item(DefId(0:6 ~ const_promotion_extern_static[317d]::BAR[0])), promoted: None }
+  // pass_name = PromoteTemps
+- // disambiguator = before
++ // disambiguator = after
+  
+  static mut  BAR: *const &'static i32 = {
+      let mut _0: *const &'static i32;     // return place in scope 0 at $DIR/const-promotion-extern-static.rs:8:17: 8:36
+      let mut _1: &[&'static i32];         // in scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+      let mut _2: &[&'static i32; 1];      // in scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+      let _3: [&'static i32; 1];           // in scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+      let mut _4: &'static i32;            // in scope 0 at $DIR/const-promotion-extern-static.rs:8:40: 8:42
+      let _5: &i32;                        // in scope 0 at $DIR/const-promotion-extern-static.rs:8:41: 8:42
++     let mut _6: &[&'static i32; 1];      // in scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+  
+      bb0: {
+          StorageLive(_1);                 // bb0[0]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+          StorageLive(_2);                 // bb0[1]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+-         StorageLive(_3);                 // bb0[2]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+-         StorageLive(_4);                 // bb0[3]: scope 0 at $DIR/const-promotion-extern-static.rs:8:40: 8:42
+-         StorageLive(_5);                 // bb0[4]: scope 0 at $DIR/const-promotion-extern-static.rs:8:41: 8:42
+-         _5 = const Scalar(alloc0+0): &i32; // bb0[5]: scope 0 at $DIR/const-promotion-extern-static.rs:8:41: 8:42
++         _6 = const BAR::promoted[0];     // bb0[2]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
```

Comment:
IMO the test should include the promoted, which would still have the `static` reference.

Reply:
you mean the memory of the promoted?

Reply:
We can't do that in a guaranteed fashion. Promoteds can still be generic at this point I think

Reply:
We'd print the *MIR body* of the promoted (we might actually be dumping that into a separate file right now), which would include the `alloc0+0: &i32` reference.